eclipse-theia/theia
# 2152 hg head

#### What it does
When _pulling_ to update a remote Mercurial repository the UI didn't update with the latest commit history.
Added the tip hash to the `WorkingDirectoryStatus`, the repository tracker will then _see_ a change and rise the appropriate event observed by `HgHistorySupport` (which will fire an event observed by the UI widgets).

#### How to test
Create a mercurial repo in os.mbed.com and import it. With the online compiler (for example) make a change and publish it.
Now back in mbed studio pull the changes from the remote repo: both the source file and the source history should be updated with the latest commits.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
